vignettes/SEIR_Model_Compartments.Rmd

```diff
@@ -0,0 +1,137 @@
+susceptible individual, $\lambda$. This is represented in the
+first terms on the right-hand side of $\text{d}S/ \text{d}t$ and $\text{d}I/ \text{d}t$.
+Exposed individuals become infectious at rate $\gamma$. The reciprocal value $1/ \gamma$ is
+the incubation time of the virus, i.e. how many days after exposure the individual
+becomes infectious to others. Infectious individuals can either recover or die
+because of the disease. Infectious individuals in group $i\in {a,m,s}$ recover in $1/ \beta_i$
+days, thus individuals recover at rate $\beta_i$. Similarly, infectious
+individuals die due to the infection after $1/ \mu_i$ days, making the death rate $\mu_i$.
+
+In addition to these six main ODEs, the model also keeps track the number of 
+disease-related deaths ($D$) over time
+$$\frac{\text{d}D}{\text{d}t} = - \mu_{I_a->D} * I_{a} +  \mu_{I_m->D} * I_{m} +  \mu_{I_s->D} * I_{s},$$
+
+The system is closed by specifying the initial conditions
+$$S(0) = S_0,\ E(0) = E_0,\ I(0) = I_0,\ R(0) = R_0,\ C(0) = C_0,\ D(0) = D_0,$$
+In the implementation in this package, the variables are normalized so 
+$S(t) + E(t) + I_a(t) + I_m(t) + I_s(t) + R(t) + D(t) \equiv 1$ for any given $t$.
+
+Steady states of the system described above are computed by setting the right-hand
+sides of the ODEs equal to zero. 
+
+Because we build on this basic model to develop more complex epidemiological models,
+we solve the ODEs over time instead of just solving for the steady states. The
+next section shows an example of how to use the SEIR_model class.
+
+## The `SEIaImIsR` class
+In this section we will walk through an example of how to use the `SEIaImIsR`
+class.
+
+To create a new `SEIaImIsR` object, run:
+```{r}
+model <- new("SEIaImIsR", name = "model")
+```
+
+Next, we assign values for the fractions of the population.
+The initial conditions must sum to one.
+```{r}
+S <- 0.9
+E <- 0.1
+I_asymptomatic <- 0
+I_mild <- 0
+I_severe <- 0
+R <- 0
+D_cumulative <- 0
+```
+
+Then, we define the transition rates:
+```{r}
+lam <- 1.0
+gamma <- 0.5
+omega <- 0.01
+e2i <- list(i_mild = 0.35, i_severe = 0.05)
+i2r <- list(i_asymptomatic = 0.9, i_mild = 0.5, i_severe = 0.05)
+pdeath <- list(i_asymptomatic = 0.005, i_mild = 0.05, i_severe = 0.3)
+# simulation period: 3000 time points (to relaxation)
+t <- seq(0, 3000, by = 1)
+```
+
+And, we set time for the simulation:
+```{r}
+t <- seq(0, 3000, by = 1)
+```
+
+We initialise our model with the parameters defined above:
+```{r}
+model <- set_init(object = model, S = S, E = E, I_asymptomatic = I_asymptomatic, I_mild = I_mild,
+                    I_severe = I_severe, R = R, D_cumulative = D_cumulative,
+                    lam = lam, gamma = gamma, omega = omega,
+                    e2i = e2i, i2r = i2r, pdeath = pdeath)
+```
+
+We use the ode solver and then plot the relaxation into the steady state:
+```{r}
+model <- ode_simulate(model, t)
+model <- plot_ode_output(model)
+```
+
```

Comment:
Can we show how transmission changes as we change one / more of the parameters? I'd say, once we have it, allowing  different values for lambda_a, lambda_m and lambda_s would be interesting to compare to the lambda_a=lambda_m=lambda_m case, perhaps?

Reply:
Shall we add another model in the vignette, plot and compare it?

Reply:
update: allowed different effective contact rates for each symptom compartment.